src/extensions/runtimes/common.ts

```diff
@@ -0,0 +1,129 @@
+import { logLabeledWarning } from "../../utils";
+import { Options } from "../../options";
+import { normalizeAndValidate } from "../../functions/projectConfig";
+import { loadCodebases } from "../../deploy/functions/prepare";
+import { Build, DynamicExtension } from "../../deploy/functions/build";
+import { getFirebaseConfig } from "../../functionsConfig";
+import {
+  EndpointFilter as Filter,
+  targetCodebases,
+} from "../../deploy/functions/functionsDeployHelper";
+import { logger } from "../../logger";
+
+export { DynamicExtension } from "../../deploy/functions/build";
+
+const savedLoggerSilent = (logger as any).silent;
```

Comment:
I don't like that `(logger as any).slient` is subverting the type system. Could you either change the type of logger so that this is not necessary, or implement silenceLogging() in logger.ts?

Reply:
Done. (I added silent to the Logger interface.)